azurerm/internal/services/consumption/consumption_budget_resource_group_resource.go

```diff
@@ -0,0 +1,60 @@
+package consumption
+
+import (
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/consumption/parse"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/pluginsdk"
+)
+
+func resourceArmConsumptionBudgetResourceGroup() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmConsumptionBudgetResourceGroupCreateUpdate,
+		Read:   resourceArmConsumptionBudgetResourceGroupRead,
+		Update: resourceArmConsumptionBudgetResourceGroupCreateUpdate,
+		Delete: resourceArmConsumptionBudgetDelete,
+		Importer: pluginsdk.ImporterValidatingResourceId(func(id string) error {
+			_, err := parse.ConsumptionBudgetID(id)
```

Comment:
I have concerns with this generic `ConsumptionBudgetID`. Someone could pass any consumption budget id as if it was a Consumption Budget ResourceGroup and Terraform would manage it. I saw that in the id tests, you were testing various consumption budget ids from different resources. I'd like to see that expanded out so we have `ConsumptionBudgetResourceGroupID` so we have 100% certainty  that the resource is in fact a Consumption Budget Resource Group. Do you think that's possible?

Reply:
That's a good call out and for details I pass in the the various consumption budget IDs to test the ID  across the 8 various scopes that a budget could be applied to. I do believe it's possible to expand these out into per consumption budget scope ID structs with a generator. My thought process is to make a generator for Consumption Budget ID structs that will generate the code for each ID and it could leverage existing validators to verify that that scope provided is in the format expected for that ID type

Reply:
That ID path you laid out sounds great! Have you seen how we generate IDs in other services? Do you think a format like [this](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/internal/services/compute/resourceids.go) could be used for Consumption Budget? Hopefully it won't be anymore issue then that for generating the ids.

Reply:
@mbfrahry I was able to use the available resource ID generator for this as the new features that have been added since my initial opening of this PR now support resource IDs such as the ones for Consumption Budgets. Thanks for pointing them out!